atomic_reactor/util.py

```diff
@@ -852,4 +853,226 @@ def df_parser(df_path, workflow=None, cache_content=False, env_replace=True, par
     return dfparser
 
 
+class Labels(object):
+    def __init__(self, df_labels):
+        """
+        Create a new Labels object
+        providing access to actual newest labels as well as old ones
+        """
+        self.LABEL_TYPE_NAME = 1
+        self.LABEL_TYPE_VERSION = 2
+        self.LABEL_TYPE_RELEASE = 3
+        self.LABEL_TYPE_ARCH = 4
+        self.LABEL_TYPE_VENDOR = 5
+        self.LABEL_TYPE_SOURCE = 6
+        self.LABEL_TYPE_COMPONENT = 7
+        self.LABEL_TYPE_HOST = 8
+
+        self._label_names = {
+            self.LABEL_TYPE_NAME : ('name', 'Name'),
+            self.LABEL_TYPE_VERSION : ('version', 'Version'),
+            self.LABEL_TYPE_RELEASE : ('release', 'Release'),
+            self.LABEL_TYPE_ARCH : ('architecture', 'Architecture'),
+            self.LABEL_TYPE_VENDOR : ('vendor', 'Vendor'),
+            self.LABEL_TYPE_SOURCE : ('authoritative-source-url', 'Authoritative_Registry'),
+            self.LABEL_TYPE_COMPONENT : ('com.redhat.component', 'BZComponent'),
+            self.LABEL_TYPE_HOST : ('com.redhat.build-host', 'Build_Host')
+        }
+
+        self._df_labels = df_labels
+        self._label_values = {}
+        for label_type in self._label_names:
```

Comment:
Nitpick, but a more pythonic way of doing this loop is to use `items` to avoid indexing on the next `for-loop`:

```python
for label_type, lbl_names in self._label_names.items():
  for lbl_name in lbl_names:
```